torch/_inductor/graph.py

```diff
@@ -304,7 +304,8 @@ def run_node(self, n: torch.fx.Node):
             if num_users > 1 and isinstance(result, TensorBox):
                 for user in n.users:
                     if user.target in needs_realized_inputs or user.op == "output":
```

Comment:
I think it'd probably be cleaner if we just added a separate case here.

So something like
```
if user.target in needs_realized_inputs:
    result.realize_hint()
elif user.op == 'output':
    result.realize()
```

Reply:
We'd still need a check of `isinstance(self.data, (Pointwise, Reduction))` here, but sure, I can move it here.

Reply:
Is there a case where we don't want to realize a buffer having the 'output' as the user? I saw the `num_users > 1` check here, not sure if it is needed for the output..